spk/rest-server/src/service-setup.sh

```diff
@@ -0,0 +1,50 @@
+
+PATH="${SYNOPKG_PKGDEST}/bin:${PATH}"
+GROUP="sc-rest-server"
+
+INST_ETC="/var/packages/${SYNOPKG_PKGNAME}/etc"
+INST_VARIABLES="${INST_ETC}/installer-variables"
+
+# Reload wizard variables stored by postinst
+if [ -r "${INST_VARIABLES}" ]; then
+    # we cannot source the file to reload the variables, when values have special characters like <, >, ...
+    for _line in $(cat "${INST_VARIABLES}"); do
+        _key="$(echo ${_line} | awk -F'=' '{print $1}')"
+        _value="$(echo ${_line} | awk -F'=' '{print $2}')"
+        declare "${_key}=${_value}"
+    done
+fi
+
```

Comment:
we have the function `reload_inst_variables` availabel here (from mk/spksrc.servier.installer.functions)
so you should use this function

```suggestion
# Source installer variables and functions to be available for service start (and not only during installation)
INST_FUNCTIONS=$(dirname $0)"/functions"
if [ -r "${INST_FUNCTIONS}" ]; then
    . "${INST_FUNCTIONS}"
fi

reload_inst_variables
```

_but it is not available in upgrade_uifile.sh, there you have to keep it_

Reply:
It is probably not a god idea, to compose the SERVICE_COMMAND from wizard_variables. 

what about writing the ARGs into an options file at installation time and use this file to add additional arguments to the SERVICE_COMMAND?


There is a similar approach in spk/syncthing/src/service-setup.sh to configure additional arguments by the user and not by wizard varialbes.

Then the SERVICE_COMMAND must only be adjusted in service_prestart.

Reply:
Your example code doesn't work for me. All wizard variables are empty. Did you test it?

Reply:
I've found the problem in spksrc.service.installer.functions -> reload_inst_variables ()
```
reload_inst_variables ()
{
    # Reload wizard variables stored by postinst
    if [ -r "${INST_VARIABLES}" ]; then
        # we cannot source the file to reload the variables, when values have special characters like <, >, ...
        for _line in $(cat "${INST_VARIABLES}"); do
            _key="$(echo ${_line} | awk -F'=' '{print $1}')"
            _value="$(echo ${_line} | awk -F'=' '{print $2}')"
            declare "${_key}=${_value}"
        done
    fi
}
```

Declare needs the -g flag to make declared variables available outside the function. Should I change this or will this break something else?

Reply:
yes please, I have this issue elsewhere in the validate function...

Reply:
Fixed in https://github.com/SynoCommunity/spksrc/pull/5550/commits/65187ff24eb49c68aab9e62f1a964d023c7b320b. I don't see any benefits in adding another config file for ARGS. The way it is everything is in one place using existing code. Adding another file just adds complexity.